### PR TITLE
Initialize frontend

### PR DESCRIPTION
**Pull request by edgeless-network**
- Configure Privy for wallets/accounts
- Add blockchain functionality as button components, just pass in the parameters to initialize the button and click the button to do the functionality
- Add blockchain related helpers

### CODE REVIEW

File: packages/frontend/.env.local
Comment (csirak1528) [inline packages/frontend/.env.local:1]: .gitignore lol

File: packages/frontend/.vscode/extensions.json
Comment (csirak1528) [inline packages/frontend/.vscode/extensions.json:1]: errors

File: packages/frontend/components/blockchain/AddFundingButton.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+}: AddFundingButtonProps) {
+	const { user, sendTransaction } = usePrivy()
+	const { wallets } = useWallets()
+	const embeddedWallet = wallets.find(
+		(wallet) => wallet.walletClientType === 'privy',
+	)
+	const chainId = Number(embeddedWallet?.chainId || 202)
+
+	const [mintCollateralTokenTx, setMintCollateralTokenTx] =
+		useState<UnsignedTransactionRequest | null>(null)
```
Comment (csirak1528) [inline packages/frontend/components/blockchain/AddFundingButton.tsx:35]: define option type instead ?

File: packages/frontend/components/blockchain/AddFundingButton.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	const [approveCollateralTokenTx, setApproveCollateralTokenTx] =
+		useState<UnsignedTransactionRequest | null>(null)
+	const [addLiquidityTx, setAddLiquidityTx] =
+		useState<UnsignedTransactionRequest | null>(null)
+
+	useEffect(() => {
+		const fetchData = async () => {
+			try {
+				const result = await deposit(chainId, amount)
+				setMintCollateralTokenTx(result)
```
Comment (csirak1528) [inline packages/frontend/components/blockchain/AddFundingButton.tsx:45]: use .catch ?

Comment (csirak1528) [reply packages/frontend/components/blockchain/AddFundingButton.tsx:45]: and .then no need for async internal useeffect

File: packages/frontend/components/blockchain/BuyConditionalTokensButton.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		useState<UnsignedTransactionRequest | null>(null)
+	const [approveCollateralTokenTx, setApproveCollateralTokenTx] =
+		useState<UnsignedTransactionRequest | null>(null)
+	const [buyTx, setBuyTx] = useState<UnsignedTransactionRequest | null>(null)
+
+	useEffect(() => {
+		const fetchData = async () => {
+			try {
+				const result = await deposit(chainId, buyAmount)
+				setMintCollateralTokenTx(result)
```
Comment (csirak1528) [inline packages/frontend/components/blockchain/BuyConditionalTokensButton.tsx:45]: apply comment ^^ to all cases

File: packages/frontend/components/blockchain/BuyConditionalTokensButton.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+	if (!user) return <button disabled={true}>Log in</button>
+
+	// Users must have an embedded wallet at `user.wallet` to send a transaction.
+	return (
+		<>
+			<button
+				disabled={!user.wallet}
+				onClick={async () => {
+					if (!mintCollateralTokenTx || !approveCollateralTokenTx || !buyTx)
+						return
```
Comment (csirak1528) [inline packages/frontend/components/blockchain/BuyConditionalTokensButton.tsx:87]: throw error here or also disable

File: packages/frontend/components/blockchain/CreateGameButton.tsx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+	question,
+	outcomes,
+	collateralToken,
+}: CreateGameButtonProps) {
+	const { user, sendTransaction } = usePrivy()
+	const { wallets } = useWallets()
+	const embeddedWallet = wallets.find(
+		(wallet) => wallet.walletClientType === 'privy',
+	)
+	const chainId = Number(embeddedWallet?.chainId || 202)
```
Comment (csirak1528) [inline packages/frontend/components/blockchain/CreateGameButton.tsx:40]: why not just use bignumber...

Comment (csirak1528) [reply packages/frontend/components/blockchain/CreateGameButton.tsx:40]: or wtv base type is

File: packages/frontend/src/abis/ConditionalTokens.tsx
Comment (csirak1528) [inline packages/frontend/src/abis/ConditionalTokens.tsx:1]: no need for abis to be tsx

File: packages/frontend/src/blockchain/weth.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import { encodeFunctionData } from 'viem'
+import { WETH_ABI } from '../abis/WETH'
+
+/**
+ * @dev Approve a spender to spend Weth
+ * @param to Address to approve
+ * @param amount Amount to approve
+ * @returns UnsignedTransactionRequest to be passed into Privy
+ */
+export const approve = async (
```
Comment (csirak1528) [inline packages/frontend/src/blockchain/weth.tsx:14]: not that big of a deal but i dont like the generic naming but it works for now.

File: packages/frontend/components/blockchain/AddFundingButton.tsx (hunk 2 above)
Comment (edgeless-network) [reply packages/frontend/components/blockchain/AddFundingButton.tsx:45]: What's the benefit of doing it this way? What is the tradeoff of not using useEffect

File: packages/frontend/components/blockchain/CreateGameButton.tsx (hunk 5 above)
Comment (edgeless-network) [reply packages/frontend/components/blockchain/CreateGameButton.tsx:40]: Don't think this really matters, too much work to change

File: packages/frontend/src/abis/ConditionalTokens.tsx
Comment (edgeless-network) [reply packages/frontend/src/abis/ConditionalTokens.tsx:1]: fixed

File: packages/frontend/components/blockchain/AddFundingButton.tsx (hunk 1 above)
Comment (edgeless-network) [reply packages/frontend/components/blockchain/AddFundingButton.tsx:35]: not sure what you mean

**Pull request merged**